### PR TITLE
Add Contra shield

### PR DESCRIPTION
**Pull request by iangus**
* Adds shield configuration for the [Contra PCB](https://github.com/ai03-2725/Contra)
* Tested on nice!nano

### CODE REVIEW

File: app/boards/shields/contra/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2020 The ZMK Contributors
```
Comment (Nicell) [inline app/boards/shields/contra/Kconfig.defconfig:1]:
```suggestion
# Copyright (c) 2021 The ZMK Contributors
```

File: app/boards/shields/contra/Kconfig.shield [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Copyright (c) 2020 The ZMK Contributors
```
Comment (Nicell) [inline app/boards/shields/contra/Kconfig.shield:1]:
```suggestion
# Copyright (c) 2021 The ZMK Contributors
```
Please change the rest of the headers too. I won't explicitly mark the rest.

**Review by Nicell — CHANGES_REQUESTED** [review]
@iangus This looks good! Clean and simple shield by the looks of it. Could you update the copyright year please?

**Comment by iangus** [comment]
@Nicell updated the copyright headers to 2021 with ceee2ec

**Comment by Nicell** [comment]
@iangus Thanks for the quick response. We unfortunately have a long queue of shields we're working through, which requires a rebase after each one is merged in. I'll ping you again here when you're up for bat and we need you to rebase/make final changes.

**Comment by iangus** [comment]
@Nicell sounds good. Just mention me here or in the discord, and I'll get this rebased/ready for merge.

**Comment by petejohanson** [comment]
Thank you, contributor, for your patience with how long review and merge of boards/shields has taken!

There are three recent refactors/changes to boards and shields that require some attention, and then we can finally get this PR merged!

1. Hardware Metadata
1. Pro Micro shield DT naming changes
1. Split changes for BLE advertising

# Hardware Metadata

## The Problem

When first developing the process around contributing new shields/boards to ZMK, we failed to recognize that several key files (setup scripts, documentation page of supported hardware, and GH Action `build.yml` file) required changes, often in the same spot, for every PR. This resulted in immediate merge conflicts for every *other* PR after one was merged, which is a  headache for contributors.

## The Fix

By adding discrete metadata files that are located with the boards/shields in question, and using that metadata to generate setup scripts, website hardware list, etc., users can contributing new hardware descriptions without the need to change the same files that other contributors are changing.

## Next Steps

First, refer to https://zmk.dev/docs/development/hardware-metadata-files to familiarize yourself with the new metadata file format.

Next, you have two options for fixing up your PR:

1. If comfortable with `git rebase`, perform an interactive rebase and revert any changes to `build.yml`, `hardware.md`, or the setup scripts `setup.sh`/`setup.ps1`, and then add the new metadata YAML file. Then force push your branch. Or,
2. Create a new branch from an up-to-date `main`, copy in the files for your new hardware, add the metadata file, then commit and push the new branch. Then, edit your open PR to point to your new branch.

# Pro Micro shield DT naming changes

In #876, we have simplified the DT naming for the "nexus node" we expose for pro-micro compatible boards, deprecating the use of `pro_micro_a`, and renaming `pro_micro_d` node to simply `pro_micro`. For pro-micro boards and shields, you'll need to adjust your DT to use the proper names.

Please see https://zmk.dev/docs/development/new-shield#shield-overlays for the updated docs on this.

# Split Shield Advertising Changes

In addition, if this is a split PR, please see #658 where we have changed our conventions to *remove* the the name from the right sides, to prevent users attempting to pair with them and causing split sync issues. This also includes removing the " Left" suffix from the naming on the left side. See the changes in that PR for examples of what to change with your split shield.

# Getting Help

If you have any questions about any of these changes, please comment here and tag @zmkfirmware/boards-shields or ask in the `#boards-shields` Discord channel.

**Review by Nicell — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**